extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java

```diff
@@ -117,6 +119,30 @@ public void setUp() throws Exception {
         instance().getConfig().addMapConfig(mapConfig);
     }
 
+    @Test
+    public void testLoadAllKeys() {
+        MapStoreConfig mapStoreConfig = new MapStoreConfig();
+        mapStoreConfig.setClassName(GenericMapStore.class.getName());
+        mapStoreConfig.setProperty(DATA_CONNECTION_REF_PROPERTY, TEST_DATABASE_REF);
+        mapStoreConfig.setProperty(TABLE_NAME_PROPERTY, tableName);
+        mapStoreConfig.setProperty(TYPE_NAME_PROPERTY, "org.example.Person");
+        mapStoreConfig.setProperty(LOAD_ALL_KEYS_PROPERTY, "false");
+
+        String mapName = tableName + "_disabled";
+        MapConfig mapConfig = new MapConfig(mapName);
+
+        mapConfig.setMapStoreConfig(mapStoreConfig);
+        instance().getConfig().addMapConfig(mapConfig);
+
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(mapName);
```

Comment:
At this point `map.size()` should return 0, could you add the assert.

Reply:
rev: 406fc8e520aaa338af760704b54aab627c851373